core/kamon-core/src/main/resources/reference.conf

```diff
@@ -241,6 +241,29 @@ kamon {
       }
     }
 
+    # Change the sampling decision for a trace automatically based on the either the number of errors in a trace, or a
+    # fixed latency threshold. The tail sampler will only work when the `span-reporting-delay` setting is set to a value
+    # greater than zero.
+    #
+    # Important things to keep in mind:
+    #   - The sampling decision override happens only in the current process. This means that any requests sent to other
+    #     services before the sampling decision override happened might be missing from the resulting trace.
+    #   - Try to set `span-reporting-delay` to a value slightly higher than your application's timeout to ensure all the
+    #     local spans related to the trace will still be kept in memory.
+    #
+    tail-sampler {
+
+      # Turns the tail sampler logic on or off.
+      enabled = no
```

Comment:
We should add a warning log if users enabled tail sampling but didn't set a `span-reporting-delay` value. The features are independent, but it is very unlikely to catch full traces without the reporting delay.